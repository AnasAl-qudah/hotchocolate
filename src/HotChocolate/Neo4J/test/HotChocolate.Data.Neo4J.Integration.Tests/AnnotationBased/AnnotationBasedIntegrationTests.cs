using CookieCrumble;
using HotChocolate.Data.Neo4J.Testing;
using HotChocolate.Execution;

namespace HotChocolate.Data.Neo4J.Integration.Tests.AnnotationBased;

[Collection(Neo4JDatabaseCollectionFixture.DefinitionName)]
public class AnnotationBasedIntegrationTests : IClassFixture<Neo4JFixture>
{
    private readonly Neo4JDatabase _database;
    private readonly Neo4JFixture _fixture;

    public AnnotationBasedIntegrationTests(Neo4JDatabase database, Neo4JFixture fixture)
    {
        _database = database;
        _fixture = fixture;
    }

    [Fact]
    public async Task MoviesSchemaIntegrationTests_GetSchema()
    {
        var tester = await _fixture.CreateSchema(_database);
        tester.Schema.MatchSnapshot();
    }

    [Fact]
    public async Task MoviesSchemaIntegrationTests()
    {
        // arrange
        var tester = await _fixture.CreateSchema(_database);

        // act
        var res1 = await tester.ExecuteAsync(@"{
                actors {
                    items {
                        name
                        actedIn {
                            title
                        }
                    }
                }
            }");

        var res2 = await tester.ExecuteAsync(@"{
                actors (where : {name : { startsWith : ""Keanu"" }}) {
                    items {
                        name
                        actedIn {
                            title
                        }
                    }
                }
            }");

        var res3 = await tester.ExecuteAsync(@"{
                movies {
                    items {
                        title
                    }
                }
            }");

        var res4 = await tester.ExecuteAsync(@"{
                actors(order: [{ name : ASC }]) {
                    items {
                        name
                        actedIn {
                            title
                        }
                    }
                }
            }");

        var res5 = await tester.ExecuteAsync(@"{
                actors(order: [{ name : ASC }] skip: 1 take: 2) {
                    items {
                        name
                        actedIn {
                            title
                        }
                    }
                }
            }");

        // assert
        await Snapshot.Create()
            .AddResult(res1, "MoviesSchema_Actors_Query")
            .AddResult(res2, "MoviesSchema_Name_StartsWith_Actors_Query")
            .AddResult(res3, "MoviesSchema_Movies_Query")
            .AddResult(res4, "MoviesSchema_Name_Desc_Sort_Actors_Query")
            .AddResult(res5, "MoviesSchema_Name_Desc_Sort_Actors_SkipOne_TakeTwo_Query")
            .MatchAsync();
    }
}
