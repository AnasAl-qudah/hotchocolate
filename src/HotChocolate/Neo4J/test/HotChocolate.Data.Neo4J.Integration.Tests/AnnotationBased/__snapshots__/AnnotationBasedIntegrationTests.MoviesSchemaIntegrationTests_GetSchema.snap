schema {
  query: Query
}

type Actor {
  name: String!
  actedIn: [Movie!]!
}

"A segment of a collection."
type ActorsCollectionSegment {
  "Information to aid in pagination."
  pageInfo: CollectionSegmentInfo!
  "A flattened list of the items."
  items: [Actor!]
}

"Information about the offset pagination."
type CollectionSegmentInfo {
  "Indicates whether more items exist following the set defined by the clients arguments."
  hasNextPage: Boolean!
  "Indicates whether more items exist prior the set defined by the clients arguments."
  hasPreviousPage: Boolean!
}

type Movie {
  title: String!
  genres: [Actor!]!
}

"A segment of a collection."
type MoviesCollectionSegment {
  "Information to aid in pagination."
  pageInfo: CollectionSegmentInfo!
  "A flattened list of the items."
  items: [Movie!]
}

type Query {
  actors(skip: Int take: Int where: ActorFilterInput order: [ActorSortInput!]): ActorsCollectionSegment
  movies(skip: Int take: Int where: MovieFilterInput order: [MovieSortInput!]): MoviesCollectionSegment
}

input ActorFilterInput {
  and: [ActorFilterInput!]
  or: [ActorFilterInput!]
  name: StringOperationFilterInput
  actedIn: ListFilterInputTypeOfMovieFilterInput
}

input ActorSortInput {
  name: SortEnumType
}

input ListFilterInputTypeOfActorFilterInput {
  all: ActorFilterInput
  none: ActorFilterInput
  some: ActorFilterInput
  any: Boolean
}

input ListFilterInputTypeOfMovieFilterInput {
  all: MovieFilterInput
  none: MovieFilterInput
  some: MovieFilterInput
  any: Boolean
}

input MovieFilterInput {
  and: [MovieFilterInput!]
  or: [MovieFilterInput!]
  title: StringOperationFilterInput
  genres: ListFilterInputTypeOfActorFilterInput
}

input MovieSortInput {
  title: SortEnumType
}

input StringOperationFilterInput {
  and: [StringOperationFilterInput!]
  or: [StringOperationFilterInput!]
  eq: String
  neq: String
  contains: String
  ncontains: String
  in: [String]
  nin: [String]
  startsWith: String
  nstartsWith: String
  endsWith: String
  nendsWith: String
}

enum SortEnumType {
  ASC
  DESC
}

"The `@defer` directive may be provided for fragment spreads and inline fragments to inform the executor to delay the execution of the current fragment to indicate deprioritization of the current fragment. A query with `@defer` directive will cause the request to potentially return multiple responses, where non-deferred data is delivered in the initial response and data deferred is delivered in a subsequent response. `@include` and `@skip` take precedence over `@defer`."
directive @defer("If this argument label has a value other than null, it will be passed on to the result of this defer directive. This label is intended to give client applications a way to identify to which fragment a deferred result belongs to." label: String "Deferred when true." if: Boolean) on FRAGMENT_SPREAD | INLINE_FRAGMENT

"The `@stream` directive may be provided for a field of `List` type so that the backend can leverage technology such as asynchronous iterators to provide a partial list in the initial response, and additional list items in subsequent responses. `@include` and `@skip` take precedence over `@stream`."
directive @stream("If this argument label has a value other than null, it will be passed on to the result of this stream directive. This label is intended to give client applications a way to identify to which fragment a streamed result belongs to." label: String "The initial elements that shall be send down to the consumer." initialCount: Int! = 0 "Streamed when true." if: Boolean) on FIELD